ktor-client/ktor-client-cio/jvmAndNix/src/io/ktor/client/engine/cio/Endpoint.kt

```diff
@@ -159,7 +216,7 @@ internal class Endpoint(
                 }
 
                 val socket = when (connectTimeout) {
-                    HttpTimeout.INFINITE_TIMEOUT_MS -> connect()
+                    HttpTimeout.INFINITE_TIMEOUT_MS -> coroutineScope { connect() }
```

Comment:
why this change?

Reply:
It was a warning and [compilation error](https://ktor.teamcity.com/buildConfiguration/Ktor_JPMSCheck/187189?hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandBuildProblemsSection=true&expandBuildChangesSection=true) because of it

Reply:
It looks like a bug in the report, could you log the issue in Kotlin YT and add `@Suppress("UNUSED_EXPRESSION")` so we can remove `coroutineScope` here

Reply:
Created issue [KT-57967](https://youtrack.jetbrains.com/issue/KT-57967)